api/src/main/java/org/apache/iceberg/expressions/BoundLiteralPredicate.java

```diff
@@ -91,6 +93,8 @@ public String toString() {
         return term() + " != " + literal;
       case STARTS_WITH:
         return term() + " startsWith \"" + literal + "\"";
+      case NOT_STARTS_WITH:
+        return term() + " notStartsWith \"" + literal + "\"";
```

Comment:
Again, I chose to go with the existing style of the code, but it does seem that the string representation of these predicates has too many quotes.

Here's an example: `ref(name="data") startsWith ""ab""`. I would expect that to be `ref(name="data") startsWith "ab"`.

It gets even weirder when using SQL, as I saw Spark showing `data NOT LIKE '"b"%'` in the explain output, though the predicate did properly evaluate to the expected `data NOT LIKE 'b%'`.

Reply:
I opened an issue for this as well, https://github.com/apache/iceberg/issues/2059. I can change both STARTS_WITH and NOT_STARTS_WITH if we think that there are in fact too many quotes in the string representation.

Reply:
I think it's best that we follow up on this once it's merged. I'd prefer to not change too much of the existing behavior so as to give this PR the most opportunity to get approved and merged in.

Then we can evaluate whether or not the toString representation is over quoted. Thankfully, it does not appear that the toString representation of the bound literal term affects the results of the operation, at least based on my running through several of the tests with the debugger.

Reply:
I'll resolve this conversation so as to make the PR more digestible to others, feel free to open it back up if you think this needs to be dealt with _in this PR_.